### ISSUE TITLE
[Feature request] Ability to define shared secrets, such as api keys

### ISSUE THREAD

**Issue by paulrostorp** [apollographql/apollo-studio-community#148]
I would like the ability to define shared secrets, such as api keys, for use in authentication, for example

Github's implementation of secrets is a good example, wherein the value of a secret is immutable, i.e you can only ever write a secret. Once created you cannot view the value of secret, but you can use them in headers or in preflight scripts.
A system like that would be suitable for storing API keys.

**Comment by cheapsteak** [comment]
Thanks for the feedback!

Could I clarify if this is specifically asking for us to store the secrets so it can be shared with other users of your graph? (i.e. 
environment variables wouldn't suffice )

**Comment by paulrostorp** [comment]
@cheapsteak yes it is for sharing. For example for use in a preflight script requiring a secret API key.
